test/unit/concerns/orchestration/compute_test.rb

```diff
@@ -52,6 +106,21 @@ class OrchestrationComputeTest < ActiveSupport::TestCase
       assert_not_includes tasks, "Detach ISO image from CDROM drive for #{@host.name}"
     end
 
+    test 'the iso should be detached when the proxmox host leaves build mode' do
+      setup_proxmox
+      @proxmox_host.stubs(:compute?).returns(true)
+      old = stub()
+      old.stubs(:build?).returns(true)
+      @proxmox_host.stubs(:old).returns(old)
+      @proxmox_host.stubs(:build?).returns(false)
+      @proxmox_host.send(:queue_bootdisk_compute)
+      tasks = @proxmox_host.queue.all.map(&:name)
+      assert_not_includes tasks, "Generating ISO image for #{@proxmox_host.name}"
+      assert_not_includes tasks, "Upload ISO image to datastore for #{@proxmox_host.name}"
+      assert_not_includes tasks, "Attach ISO image to CDROM drive for #{@proxmox_host.name}"
+      assert_includes tasks, "Detach ISO image from CDROM drive for #{@proxmox_host.name}"
+    end
+
     test 'the iso should be detached when the host leaves build mode' do
       @host.stubs(:compute?).returns(true)
       old = stub()
```

Comment:
It would be probalbly much better to introduce a new test that only includes those new methods, but I don't insist.

Reply:
Do you mean to add only `iso_upload`, `iso_attach` and, `iso_detach` tests and remove others?

Reply:
What I mean was extracting all new methods into a separate TestClass.